src/devices/Tm1637/Tm1637.cs

```diff
@@ -0,0 +1,299 @@
+                if (value > 7)
+                    throw new ArgumentException($"{nameof(Brightness)} can't be more than 7");
+                _brightness = value;
+                DisplayRaw(0, _lastDisplay[0]);
+            }
+        }
+
+        private PinValue WriteByte(byte data)
+        {
+            // We send data by 8 bits
+            for (byte i = 0; i < 8; i++)
+            {
+
+                _controller.Write(_pinClk, PinValue.Low);
+                DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+                // LSB first
+                if ((data & 0x01) == 0x01)
+                    _controller.Write(_pinDio, PinValue.High);
+                else
+                    _controller.Write(_pinDio, PinValue.Low);
+                // LSB first
+                data >>= 1;
+                _controller.Write(_pinClk, PinValue.High);
+                DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+            }
+
+            // Wait for the acknoledge
+            _controller.Write(_pinClk, PinValue.Low);
+            _controller.Write(_pinDio, PinValue.High);
+            _controller.Write(_pinClk, PinValue.High);
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+            _controller.SetPinMode(_pinDio, PinMode.Input);
+
+            // Wait 1 µs, it's the waiting time between clk up and down
+            // That's according to the documentation
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+
+            var ack = _controller.Read(_pinDio);
+            if (ack == PinValue.Low)
+            {
+                _controller.SetPinMode(_pinDio, PinMode.Output);
+                _controller.Write(_pinDio, PinValue.Low);
+            }
+
+            _controller.Write(_pinClk, PinValue.High);
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+            _controller.Write(_pinClk, PinValue.Low);
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+
+            _controller.SetPinMode(_pinDio, PinMode.Output);
+            return ack;
+        }
+
+        private void StartTransmission()
+        {
+            _controller.Write(_pinClk, PinValue.High);
+            _controller.Write(_pinDio, PinValue.High);
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+            _controller.Write(_pinDio, PinValue.Low);
+        }
+
+        private void StopTransmission()
+        {
+            _controller.Write(_pinClk, PinValue.Low);
+            _controller.Write(_pinDio, PinValue.Low);
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+            _controller.Write(_pinClk, PinValue.High);
+            _controller.Write(_pinDio, PinValue.High);
+        }
+
+        /// <summary>
+        /// Displays segments starting at first segment with byte array containing raw data for each segment including the dot
+        /// 
+        /// bit 0 = a       _a_
```

Comment:
Perhaps this table should be in the <remarks> section of the comment?

Reply:
What do you mean by:

> Perhaps this table should be in the  section of the comment?

Reply:
sorry I meant the `<remarks>` section of the comment.